HopsanGUI/Dialogs/OptimizationDialog.cpp

```diff
@@ -568,16 +620,53 @@ void OptimizationDialog::applyParameters()
 }
 
 
-//! @brief Asks user whether or not to overwrite optimization script
-bool OptimizationDialog::checkIfScriptIsEmpty()
+//! @brief Slot for flagging that script text has changed
+void OptimizationDialog::scriptTextChanged()
 {
-    if(!mpScriptBox->document()->isEmpty())
+    mScriptTextChanged = (mpScriptBox->toPlainText() != mSavedScriptText);
```

Comment:
Why is this comparison needed, is it not enough that the signal is emitted, (textChanged)?
Or is that signal emitted to often?

Reply:
It is needed because the slot is called also when user presses ctrl-z. If all changes have been undone, the asterisk shall be removed from the tab text.

Reply:
Ok